File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/rcm/hosmiddle/d/HOSMIDDLEDO.java
package iih.ci.rcm.hosmiddle.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.rcm.hosmiddle.d.desc.HOSMIDDLEDODesc;
import java.math.BigDecimal;

/**
 * 院感上报中间表 DO数据 
 * 
 */
public class HOSMIDDLEDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_MIDDLE= "Id_middle";
	public static final String ID_ENT= "Id_ent";
	public static final String ID_HOSPITALREPORT= "Id_hospitalreport";
	public static final String ID_SUBCARD= "Id_subcard";
	public static final String TITLE= "Title";
	public static final String STATE= "State";
	public static final String STYLE= "Style";
	public static final String SERAILNO= "Serailno";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_middle() {
		return ((String) getAttrVal("Id_middle"));
	}	
	public void setId_middle(String Id_middle) {
		setAttrVal("Id_middle", Id_middle);
	}
	public String getId_ent() {
		return ((String) getAttrVal("Id_ent"));
	}	
	public void setId_ent(String Id_ent) {
		setAttrVal("Id_ent", Id_ent);
	}
	public String getId_hospitalreport() {
		return ((String) getAttrVal("Id_hospitalreport"));
	}	
	public void setId_hospitalreport(String Id_hospitalreport) {
		setAttrVal("Id_hospitalreport", Id_hospitalreport);
	}
	public String getId_subcard() {
		return ((String) getAttrVal("Id_subcard"));
	}	
	public void setId_subcard(String Id_subcard) {
		setAttrVal("Id_subcard", Id_subcard);
	}
	public String getTitle() {
		return ((String) getAttrVal("Title"));
	}	
	public void setTitle(String Title) {
		setAttrVal("Title", Title);
	}
	public FBoolean getState() {
		return ((FBoolean) getAttrVal("State"));
	}	
	public void setState(FBoolean State) {
		setAttrVal("State", State);
	}
	public String getStyle() {
		return ((String) getAttrVal("Style"));
	}	
	public void setStyle(String Style) {
		setAttrVal("Style", Style);
	}
	public String getSerailno() {
		return ((String) getAttrVal("Serailno"));
	}	
	public void setSerailno(String Serailno) {
		setAttrVal("Serailno", Serailno);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_middle";
	}
	
	@Override
	public String getTableName() {	  
		return "CI_MR_HOS_MIDDLE";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(HOSMIDDLEDODesc.class);
	}
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.rcm/hosmiddle/d/HOSMIDDLEDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.rcm.hosmiddle.d
{
    /// <summary>
    /// HOSMIDDLEDO 的摘要说明。
    /// </summary>
    public class HOSMIDDLEDO : BaseDO {

		public const string TABLE_NAME = "CI_MR_HOS_MIDDLE";
		public const string TABLE_ALIAS_NAME = "a0";

        public HOSMIDDLEDO() {
            this.State = true;
        }
		public string Id_middle {
            get { return getAttrVal<string>("Id_middle",null); }
            set { setAttrVal<string>("Id_middle", value); }
        }
		public string Id_ent {
            get { return getAttrVal<string>("Id_ent",null); }
            set { setAttrVal<string>("Id_ent", value); }
        }
		public string Id_hospitalreport {
            get { return getAttrVal<string>("Id_hospitalreport",null); }
            set { setAttrVal<string>("Id_hospitalreport", value); }
        }
		public string Id_subcard {
            get { return getAttrVal<string>("Id_subcard",null); }
            set { setAttrVal<string>("Id_subcard", value); }
        }
		public string Title {
            get { return getAttrVal<string>("Title",null); }
            set { setAttrVal<string>("Title", value); }
        }
		public bool State {
            get { return getAttrVal<FBoolean>("State",true); }
            set { setAttrVal<FBoolean>("State", value); }
        }
		public string Style {
            get { return getAttrVal<string>("Style",null); }
            set { setAttrVal<string>("Style", value); }
        }
		public string Serailno {
            get { return getAttrVal<string>("Serailno",null); }
            set { setAttrVal<string>("Serailno", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "CI_MR_HOS_MIDDLE";
        }
        
        /// <summary>
        /// 返回表别名
        /// </summary>
        /// <returns></returns>
        public override string getAliasTableName()
        {
            return "a0";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_middle";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.rcm.hosmiddle.d.HOSMIDDLEDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_middle", "Id_ent", "Id_hospitalreport", "Id_subcard", "Title", "State", "Style", "Serailno"};
        }
        
    }
}
